src/drm.cpp

```diff
@@ -1887,6 +1902,32 @@ bool drm_set_connector( struct drm_t *drm, struct connector *conn )
 	drm->connector = conn;
 	drm->needs_modeset = true;
 
+	if (conn->props.count("panel orientation") > 0 && g_drmModeOrientation == PANEL_ORIENTATION_AUTO) 
+	{
+		const char *orientation = get_enum_name(conn->props["panel orientation"], conn->initial_prop_values["panel orientation"]);
+
+		if (strcmp(orientation, "Normal") == 0)
+		{
+			g_drmModeOrientation = PANEL_ORIENTATION_0;
```

Comment:
This overwrites `g_drmModeOrientation`, which means the next time `drm_set_connector()` is called, the orientation of the panel will be retained.

The idea is to update `g_drmEffectiveOrientation` here, not `g_drmModeOrientation`. That way `g_drmModeOrientation` always reflects what the user wants (CLI flag), and `g_drmEffectiveOrientation` always reflects what's been decided depending on the KMS prop and heuristics.